crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -260,6 +260,18 @@ impl<'db> TypeInferenceBuilder<'db> {
             NodeWithScopeKind::FunctionTypeParameters(function) => {
                 self.infer_function_type_params(function.node());
             }
+            NodeWithScopeKind::ListComprehension(comprehension) => {
+                self.infer_list_comprehension_expression_in_scope(comprehension.node());
```

Comment:
I prefer these method names without the word "in", e.g. `infer_list_comprehension_expression_scope`. Using "in" almost seems to communicate the opposite of what we want. What distinguishes these methods is that they infer the comprehension scope, not the comprehension in its containing scope.